### ISSUE TITLE
Allow setting a cloud variable to the current username

### ISSUE THREAD

**Issue by rdococ** [LLK/scratch-vm#2057]
You should be able to set a cloud variable to the username of the person that is currently viewing the project.

Setting a cloud variable to any other string, including any other username, will not work - only setting the cloud variable to the current username, gotten with the (username) block, would work.

This would make recording high scores much easier than it is currently, without removing too many restrictions from cloud variables and making them too difficult to moderate.

**Comment by towerofnix** [comment]
Why not have a block like "user who most recently changed (cloud variable)"? It sounds like it would have the same benefits to me!

**Comment by griffpatch** [comment]
Yes, me, theLogFather and many others want this block (but perhaps a drop
down version that also gives the 'time' the block was last updated (in
server time perhaps)

**Comment by TheLogFather** [comment]
Yes, "time since [cloudvar v] changed" (given in seconds –and note this form avoids any questions about time-zones and time format), and something like "user changed [cloudvar v]" (which would return blank if project never connected to cloud, e.g. for a New Scratcher –and note that would also provide a much less tedious and more easily comprehensible test that a project has made an initial cloud connection than the only method that currently exists, which requires setting a cloudvar to some specific value every time you save the project, and then immediately changing it in the cloud *without* the project then saving again).

**Comment by adazem009** [comment]
> Yes, "time since [cloudvar v] changed" (given in seconds –and note this form avoids any questions about time-zones and time format), and something like "user changed [cloudvar v]" (which would return blank if project never connected to cloud, e.g. for a New Scratcher –and note that would also provide a much less tedious and more easily comprehensible test that a project has made an initial cloud connection than the only method that currently exists, which requires setting a cloudvar to some specific value every time you save the project, and then immediately changing it in the cloud _without_ the project then saving again).

It's probably a bit late now but... you don't have to avoid saving the project. I do it like this:
1. Set the cloud variable to any number
2. Set the cloud variable to some string, e. g. "abc"
3. Save the project

Only the number will be saved on the cloud server.

**Comment by apumpleby** [comment]
> It's probably a bit late now but... you don't have to avoid saving the project. I do it like this:
>     1. Set the cloud variable to any number
>     2. Set the cloud variable to some string, e. g. "abc"
>     3. Save the project
> Only the number will be saved on the cloud server.

Didn't used to work like that back then, unfortunately.
IIRC, trying to set a cloudvar to an invalid string would end up just setting it to zero everywhere, both in cloud & in project.
(Which was a fairly regular cause of problems in cloud-based projects - if it messed up some sort of encoding and tried setting to something invalid, all the info it had contained was gone and users would have no idea what just happened nor why...)

**Comment by rdococ** [comment]
> Why not have a block like "user who most recently changed (cloud variable)"? It sounds like it would have the same benefits to me!

Alright, this is way better. Support for 'last user to change [cloud var]' :)
